### PR TITLE
Add Debian 12 "bookworm" support

### PR DESCRIPTION
**Pull request by kenyon**
- **metadata.json: drop EoL Debian 10 "buster"** (#1177)
- **metadata.json: add Debian 12 "bookworm"**

### CODE REVIEW

**Comment by kenyon** [comment]
Not sure what needs to happen to resolve the agent installation failure on Debian 12 acceptance tests:

```
Installation of agent failed.
Results:
  1719e17324c7: {"_output"=>"23:44:29 +0000 INFO: Version parameter not defined and no agent detected. Assuming latest.\n23:44:29 +0000 INFO: Downloading Puppet latest for Debian...\n23:44:29 +0000 INFO: Debian platform! Lets get you a DEB...\n23:44:29 +0000 INFO: Downloading [http://nightlies.puppet.com/apt/puppet8-nightly-release-.deb\n23:44:29](http://nightlies.puppet.com/apt/puppet8-nightly-release-.deb/n23:44:29) +0000 INFO:   to file /tmp/install.sh.828.290040416/puppet8-nightly-release-.deb\n23:44:29 +0000 INFO: Trying wget...\n23:44:29 +0000 INFO: Retrying... [1/5]\n23:44:29 +0000 INFO: Return code: 8\n23:44:30 +0000 INFO: Retrying... [2/5]\n23:44:30 +0000 INFO: Return code: 8\n23:44:31 +0000 INFO: Retrying... [3/5]\n23:44:31 +0000 INFO: Return code: 8\n23:44:32 +0000 INFO: Retrying... [4/5]\n23:44:32 +0000 INFO: Return code: 8\n23:44:33 +0000 INFO: Retrying... [5/5]\n23:44:33 +0000 INFO: Return code: 8\n23:44:34 +0000 CRIT: ERROR 404\n23:44:34 +0000 CRIT: Unable to retrieve a valid package!\n", "_error"=>{"kind"=>"puppetlabs.tasks/task-error", "issue_code"=>"TASK_ERROR", "msg"=>"The task failed with exit code 1", "details"=>{"exit_code"=>1}}}}
/home/runner/work/puppetlabs-apt/puppetlabs-apt/vendor/bundle/ruby/2.7.0/gems/puppet_litmus-1.4.0/lib/puppet_litmus/rake_helper.rb:335:in `raise_bolt_errors'
/home/runner/work/puppetlabs-apt/puppetlabs-apt/vendor/bundle/ruby/2.7.0/gems/puppet_litmus-1.4.0/lib/puppet_litmus/rake_helper.rb:156:in `install_agent'
/home/runner/work/puppetlabs-apt/puppetlabs-apt/vendor/bundle/ruby/2.7.0/gems/puppet_litmus-1.4.0/lib/puppet_litmus/rake_tasks.rb:130:in `block (2 levels) in <top (required)>'
/home/runner/work/puppetlabs-apt/puppetlabs-apt/vendor/bundle/ruby/2.7.0/gems/rake-13.2.1/exe/rake:27:in `<top (required)>'
/opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `load'
/opt/hostedtoolcache/Ruby/2.7.8/x64/bin/bundle:23:in `<main>'
Tasks: TOP => litmus:install_agent
```

**Comment by smortex** [comment]
```
http://nightlies.puppet.com/apt/puppet8-nightly-release-.deb
                                                        ^~~~
```

Should be something like:

```
http://nightlies.puppet.com/apt/puppet8-nightly-release-bookworm.deb
```

I have not the time to test ATM, but I think this is caused by:
https://github.com/puppetlabs/puppetlabs-apt/blob/68c444c67faebde6ca85b0f4f0e542d7cc991372/.fixtures.yml#L6-L8

which does not include (added in 4.19.0):
https://github.com/puppetlabs/puppetlabs-puppet_agent/commit/f99be1973b14a8e7e8210d079df38d1ebb393c12

Can you give it a try?

File: .fixtures.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ fixtures:
     facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
     puppet_agent:
       repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
-      ref: v4.13.0
+      ref: v4.19.0
     provision: 'https://github.com/puppetlabs/provision.git'
```
Comment (bastelfreak) [inline .fixtures.yml:9]: IMO this is a really bad design pattern and we should pull from the main branch instead

Comment (kenyon) [reply .fixtures.yml:9]: Agreed. It was like that when added in https://github.com/puppetlabs/puppetlabs-apt/commit/43a4d26621e807a1389c18e09d588f54d3f08727 but changed to a version in https://github.com/puppetlabs/puppetlabs-apt/commit/89721b16e7dcfe9cc37e082e63f70e63c7e1f2c6. 🤷

Comment (kenyon) [reply .fixtures.yml:9]: Made #1181 to do this.

**Review by smortex — APPROVED** [review]
Nice, I guess some work on the PR may be needed depending on the merge ordering, but this is great!

**Review by Ramesh7 — APPROVED** [review]

**Pull request merged**